fix: use anonymous dataset to create `remote_function`

shobsi opened the pull request:
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated https://screenshot.googleplex.com/35fii36HaJTBv6t

Fixes internal issue 308814432 🦕

shobsi committed:
fix: use anonymous dataset to create `remote_function`

jiaxunwu reviewed (changes requested):
Any doc changes needed for this?

shobsi committed:
update README about anonymous dataset instead of bigframes_temp_location

shobsi commented:
> Any doc changes needed for this?

Good call, thanks! Updated README.rst

shobsi committed:
Merge remote-tracking branch 'refs/remotes/github/main' into rf-anon-dataset

shobsi committed:
remove dataset creation step from remote function

This is because now the dataset is an anonymous dataset that must have
been created previously as part of bigframes session creation.

shobsi committed:
Merge remote-tracking branch 'refs/remotes/github/main' into rf-anon-dataset

shobsi committed:
restore create_dataset, guarded by get_dataset

shobsi commented:
`nox -s e2e` with additional `-k test_remote_` was run locally to verify large remote function tests pass.
